test/e2e_node/container_metrics_test.go

```diff
@@ -0,0 +1,147 @@
+			}
+			appendMatchesForContainer(f.Namespace.Name, pod0, pod1, "busybox-container", ctrMatches, keys, gstruct.AllowDuplicates|gstruct.IgnoreExtras)
+
+			ctrOptionalMatches := map[string]types.GomegaMatcher{
+				"container_fs_io_current":                     boundedSample(0, 100),
+				"container_fs_io_time_seconds_total":          boundedSample(0, 100),
+				"container_fs_io_time_weighted_seconds_total": boundedSample(0, 100),
+				"container_fs_inodes_free":                    boundedSample(0, 10*e2evolume.Kb),
+				"container_fs_inodes_total":                   boundedSample(0, 100),
+				"container_fs_limit_bytes":                    boundedSample(100*e2evolume.Mb, 10*e2evolume.Tb),
+				"container_fs_usage_bytes":                    boundedSample(0, 1000000),
+				"container_fs_read_seconds_total":             preciseSample(0),
+				"container_fs_reads_merged_total":             preciseSample(0),
+				"container_fs_sector_reads_total":             preciseSample(0),
+				"container_fs_sector_writes_total":            preciseSample(0),
+				"container_fs_write_seconds_total":            preciseSample(0),
+				"container_fs_writes_merged_total":            preciseSample(0),
+			}
+			// Missing from containerd, so set gstruct.IgnoreMissing
+			// See https://github.com/google/cadvisor/issues/2785
+			appendMatchesForContainer(f.Namespace.Name, pod0, pod1, "busybox-container", ctrOptionalMatches, keys, gstruct.AllowDuplicates|gstruct.IgnoreMissing|gstruct.IgnoreExtras)
+
+			podMatches := map[string]types.GomegaMatcher{
+				"container_network_receive_bytes_total":            boundedSample(10, 10*e2evolume.Mb),
+				"container_network_receive_errors_total":           boundedSample(0, 1000),
+				"container_network_receive_packets_dropped_total":  boundedSample(0, 1000),
+				"container_network_receive_packets_total":          boundedSample(0, 1000),
+				"container_network_transmit_bytes_total":           boundedSample(10, 10*e2evolume.Mb),
+				"container_network_transmit_errors_total":          boundedSample(0, 1000),
+				"container_network_transmit_packets_dropped_total": boundedSample(0, 1000),
+				"container_network_transmit_packets_total":         boundedSample(0, 1000),
+			}
+			// TODO: determine why these are missing from containerd but not CRI-O
+			appendMatchesForContainer(f.Namespace.Name, pod0, pod1, "POD", podMatches, keys, gstruct.AllowDuplicates|gstruct.IgnoreMissing|gstruct.IgnoreExtras)
+
+			matchResourceMetrics := gstruct.MatchKeys(gstruct.IgnoreExtras, keys)
+			ginkgo.By("Giving pods a minute to start up and produce metrics")
+			gomega.Eventually(ctx, getContainerMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
+			ginkgo.By("Ensuring the metrics match the expectations a few more times")
+			gomega.Consistently(ctx, getContainerMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
+		})
+		ginkgo.AfterEach(func(ctx context.Context) {
```

Comment:
Although it doesn't affect the correct execution of the tests, moving `ginkgo.AfterEach` below `ginkgo.BeforeEach` will make the code clearer. This way, we'll know right away what resources we will create and which ones we will clean up after the test ends :)
```
ginkgo.BeforeEach(func(ctx context.Context) {
	createMetricsPods(ctx, f)
})
ginkgo.AfterEach(func(ctx context.Context) {
	removeMetricsPods(ctx, f)
})
```

Reply:
done